src/libs/ReportActionsUtils.ts

```diff
@@ -368,6 +375,11 @@ function shouldReportActionBeVisibleAsLastAction(reportAction: OnyxEntry<ReportA
         return false;
     }
 
+    // Ignore Report Preview as last action for LHN display if the action is deleted and is also pending for deletion.
+    if (isReportPreviewAction(reportAction) && isMessageDeleted(reportAction) && reportAction.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
+        return false;
+    }
```

Comment:
> Could you please explain a little bit this one, what to ignore ? and why we have to do it ?

Sure. [Here](https://github.com/Expensify/App/blob/e81ed8f4c548d0c1675a6d696772a914d115c2b1/src/libs/OptionsListUtils.js#L384) we use `shouldReportActionBeVisibleAsLastAction` to check if an action (in our case Report Preview action) can be considered as last Report Action for display of LHN subtitle display text for the Chat Report. 

Before our fix, the Report Preview action was deleted in offline scenarios because of which LHN would display the subtitle text of the action that was immediately previous to the deleted Report Preview. Now, since we do not remove the Report Preview in offline scenario until app comes online, the last action would turn up to be the Report Preview thereby resulting in the text message of Report Preview as subtitle in LHN. So, we need to ignore the Report Preview action as last action for LHN if the IOU Report is deleted offline. This is the reason why we added the condition here. 

So, here, we ignore the Report Preview if the message was deleted as set [here](https://github.com/rojiphil/App/blob/127b84b9267b2235fe15073394a1c278975e013a/src/libs/actions/IOU.js#L2207) and if the pending action is `delete` which we set [here](https://github.com/rojiphil/App/blob/127b84b9267b2235fe15073394a1c278975e013a/src/libs/actions/IOU.js#L2186). In all other cases, we will not ignore and show the Report Preview text in LHN.

Hope this will suffice. Let me know otherwise.

Reply:
Have responded [here](https://github.com/Expensify/App/pull/32698/files/acfb595794d57e0e5b06c810204ad8d9a3b26111#r1433460922).
Let me know what you think.